index.d.ts

```diff
@@ -68,13 +68,8 @@ export { Duration };
 
 type RowSelectionParams = {
   selectorType?: string | null;
-  onSelect?: (
-    e: React.ChangeEvent,
-    isChecked: boolean,
-    datum: Immutable.Map<string, any>,
-    key: string | number
-  ) => void;
-  rowIsSelected?: (datum: Immutable.Map<string, any>, key: string | number) => boolean;
+  onSelect?: (e: React.ChangeEvent, isChecked: boolean, datum: GenericObject, key: string | number) => void;
```

Comment:
kind of unsure if this is where we actually defined module-wide TS definitions, but I figured it is.

Reply:
You can actually make RowSelectionParams a generic and have that generic type be the `datum` type. That way you can be type safe about the return object. You'll, of course, have to thread that generic through, but it might be worth it instead of just a random object.